### PR TITLE
Remove deprecated 2.2 code

### PR DESCRIPTION
**Pull request by scottfrederick**
See #19597 

This PR includes all deprecations that were explicitly marked `since 2.2`. There are other deprecations that are not marked with a `since` that need to be evaluated for removal in 2.3.

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/health/HealthEndpointAutoConfiguration.java [hunk 1]
```diff
@@ -35,18 +35,13 @@
...
 public class HealthEndpointAutoConfiguration {
 
 	@Bean
-	@SuppressWarnings("deprecation")
-	HealthEndpointProperties healthEndpointProperties(HealthIndicatorProperties healthIndicatorProperties) {
-		HealthEndpointProperties healthEndpointProperties = new HealthEndpointProperties();
-		healthEndpointProperties.getStatus().getOrder().addAll(healthIndicatorProperties.getOrder());
-		healthEndpointProperties.getStatus().getHttpMapping().putAll(healthIndicatorProperties.getHttpMapping());
-		return healthEndpointProperties;
+	HealthEndpointProperties healthEndpointProperties() {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/health/HealthEndpointAutoConfiguration.java]: Now that this object can be created using the default constructor, I think the whole bean definition should go away and the class-level annotation should be changed to `@EnableConfigurationProperties(HealthEndpointProperties.class)`.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/LogFile.java [hunk 2]
```diff
@@ -37,22 +37,6 @@
...
-	@Deprecated
-	public static final String FILE_PROPERTY = "logging.file";
-
-	/**
-	 * The name of the Spring property that contains the directory where log files are
-	 * written.
-	 * @deprecated since 2.2.0 in favor of {@link #FILE_PATH_PROPERTY}
-	 */
-	@Deprecated
-	public static final String PATH_PROPERTY = "logging.path";
```
[the remark this answers is not in the document]
Comment (scottfrederick) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/LogFile.java]: done

**Comment by scottfrederick** [comment]
I've resolved the two comments above, updated copyrights, and split the changes into several commits aligned to a few themes.

**Comment by scottfrederick** [comment]
Added another commit to remove Joda-Time auto-configuration.

**Pull request closed without merging**